index.js

```diff
@@ -0,0 +1,106 @@
+const urlParser = require('url');
+const utils = require('./lib/utils');
+const MagnetSigner = require('./lib/signer');
+const fetch = require('request-promise');
+
+class MagnetAPIClient {
+    constructor(endpointUri, calk, secretKey) {
+        this._calk = null;
+        this._serverAddress = null;
+        this._endpoint = null;
+        this._magnetSigner = null;
+        this._proxy = null;
+        this._calk = calk;
+        this._serverAddress = urlParser.parse(endpointUri).hostname;
+        this._endpoint = urlParser.parse(endpointUri).path;
+
+        if (secretKey) {
+            this._magnetSigner = new MagnetSigner(endpointUri, secretKey);
+        }
+    }
+
+    PrepareRequest(methodName, request) {
+        //add calk to request, if it does not exist.
+        if (!this.RequestHasCalk(request)) {
+            //request is Query String
+            if (typeof request == 'string') {
+                if (request.trim() == '') {
+                    request = `calk=${this._calk}`;
+                } else {
+                    request = `calk=${this._calk}&${request}`;
+                }
+            }  else if (typeof request == 'object') { //request is an object / key - value
+                request.calk = this._calk;
+            }
+        }
+    }
+
+    GetQueryString(methodName, request, requestMethod, signRequest) {
+        let queryString;
+        if (signRequest && this._magnetSigner != undefined && this._magnetSigner != null) {
+            queryString = this._magnetSigner.GetSignedQueryString(methodName, request, requestMethod);
+        } else if (typeof request == 'object') {
+            queryString = this.ConstructQueryString(request);
+        } else if (typeof request == 'string') {
+            queryString = request;
+        }
+        return queryString;
+    }
+
+    CallWebMethod(methodName, request, requestMethod, signRequest = true) {
```

Comment:
Why make this a breaking change when the callback form could be easily supported by just adding a then to the returned fetch promise?